core/client/fs/src/main/java/alluxio/client/block/stream/GrpcDataWriter.java

```diff
@@ -96,6 +96,8 @@ public static GrpcDataWriter create(FileSystemContext context, WorkerNetAddress
       return new GrpcDataWriter(context, address, id, length, chunkSize, type, options,
           grpcClient);
     } catch (Exception e) {
+      // Close the client before releasing in order to get it removed from the pool.
+      grpcClient.close();
```

Comment:
Is there a way to tell that the gRPC channel is in a bad state without having to manually close it?

Reply:
Particular problem here is that the channel is in a good state but needs reauthentication. This is kind of a catch all fix that could end up invalidating the client in transient network problems as well. I did it this way due to cost of redesigning the blockworker client for retry support. Options that I can think of are.
- closing the client only on UnauthenticatedException.
- implementing retries into blockworker client for transient errors.

If not going with the latter, I prefer not doing the former in order to keep the behavior less complex.